From 4a578366ef7a743c42b464dd060cd4321de33eae Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Sat, 10 May 2014 14:02:55 -0700
Subject: [PATCH 018/117] atomisp_gmin_platform: Add camera_af_platform_data
 callback

The power_ctrl callback in this struct must be present or else the IMX
driver won't initialize.  Upsteam has this stubbed out for BYT anwyay,
so this implemenation is identical.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 arch/x86/platform/intel-mid/atomisp_gmin_platform.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
index 0820005..5817994 100644
--- a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
+++ b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
@@ -63,6 +63,11 @@ const struct atomisp_platform_data *atomisp_get_platform_data(void)
 }
 EXPORT_SYMBOL_GPL(atomisp_get_platform_data);
 
+static int af_power_ctrl(struct v4l2_subdev *subdev, int flag)
+{
+	return 0;
+}
+
 /*
  * Used in a handful of modules.  Focus motor control, I think.  Note
  * that there is no configurability in the API, so this needs to be
@@ -71,10 +76,17 @@ EXPORT_SYMBOL_GPL(atomisp_get_platform_data);
  * struct camera_af_platform_data {
  *     int (*power_ctrl)(struct v4l2_subdev *subdev, int flag);
  * };
+ *
+ * Note that the implementation in MCG platform_camera.c is stubbed
+ * out anyway (i.e. returns zero from the callback) on BYT.  So
+ * neither needed on gmin platforms or supported upstream.
  */
 const struct camera_af_platform_data *camera_get_af_platform_data(void)
 {
-	return NULL;
+	static struct camera_af_platform_data afpd = {
+		.power_ctrl = af_power_ctrl,
+	};
+	return &afpd;
 }
 EXPORT_SYMBOL_GPL(camera_get_af_platform_data);
 
-- 
1.9.3

